### PR TITLE
Disable the DefaultDiagnosticConsoleFormatter because of index-out-of-bounds crash

### PR DESCRIPTION
**Pull request by d-ronnqvist**
Bug/issue #, if applicable: https://github.com/apple/swift-docc/issues/678 rdar://116943708

## Summary

Always use `IDEDiagnosticConsoleFormatter` to format diagnostics to avoid a crash.

This is a smaller change than it would be to revert the PR that introduced `DefaultDiagnosticConsoleFormatter`.

## Dependencies

None.

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.

- ~[ ] Added tests~
- [x] Ran the `./bin/test` script and it succeeded
- ~[ ] Updated documentation if necessary~

### CODE REVIEW

**Comment by d-ronnqvist** [comment]
@swift-ci please test

**Comment by d-ronnqvist** [comment]
We'll hold this PR until the next Documentation Workgroup meeting where we'll discuss https://github.com/apple/swift-docc/issues/678

**Comment by Kyle-Ye** [comment]
https://github.com/apple/swift-docc/issues/678#issuecomment-1778595554

Considering #678 seems to be fixed. Can we speed up the process of the cherry-pick to release/5.10 and observe the feedback of 5.10 toolchain instead of merging this to release/5.10. @d-ronnqvist 

I'd like to know what's your thoughts on this? cc @arthurcro

**Comment by d-ronnqvist** [comment]
> Considering [#678 (comment)](https://github.com/apple/swift-docc/issues/678#issuecomment-1778595554) seems to be fixed. Can we speed up the process of the cherry-pick to release/5.10 and observe the feedback of 5.10 toolchain instead of merging this to release/5.10. @d-ronnqvist
> 
> I'd like to know what's your thoughts on this? cc @arthurcro

I'm out all of this week and likely part of next week as well so I'll leave that decision to @QuietMisdreavus and @franklinsch. In the mean time I'll mark this PR as a draft to keep it around and also indicate that we may not need it.

File: Sources/SwiftDocC/Infrastructure/Diagnostics/DiagnosticConsoleWriter.swift [hunk 1]
```diff
@@ -42,7 +42,8 @@ public final class DiagnosticConsoleWriter: DiagnosticFormattingConsumer {
         highlight: Bool? = nil
     ) {
         outputStream = stream
-        formattingOptions = options
+        // Reenable `DefaultDiagnosticConsoleFormatter` as the default formatter when the index-out-of-bounds crash is fixed. https://github.com/apple/swift-docc/issues/722
+        formattingOptions = [.formatConsoleOutputForTools]
```
Comment (patshaughnessy) [inline Sources/SwiftDocC/Infrastructure/Diagnostics/DiagnosticConsoleWriter.swift:46]: Should we also pass `formattingOptions` into the call to `makeDiagnosticFormatter` just below? Right now it looks like we pass the original `options` value into `makeDiagnosticFormatter` which appears incorrect.

However, based on the conversation here we might end up just closing this PR out anyway...

**Review by patshaughnessy — CHANGES_REQUESTED** [review]

**Comment by d-ronnqvist** [comment]
Closing this since https://github.com/apple/swift-docc/issues/678 is closed.

**Comment by Kyle-Ye** [comment]
Some note here:
Some cherry-pick fix for #678 is still open. I've tried to ping and send private message to the related owner but still got no response. It will still cause crash on latest swift/5.10 toolchain.

**Pull request closed without merging**